### PR TITLE
feat: figma plugin assets export

### PR DESCRIPTION
**Pull request by cscheffauer**
# Description

- Added new repository, including the figma plugin for assets export
  - Built with React, Vite, ESBuild & Typescript
- Added basic plugin code to grab the pages of the document, filtered by a allowlist (more work here later)
- Added basic UI, which allows selecting Icons or Illustrations and a export button (more work here later)

To test locally:
1. Run `yarn figma-plugin-assets-export dev` in the Monorepo
2. In Figma, go to Plugins -> Development -> Import plugin from manifest:
image
3. Select the `manifest.json` from the plugin package
4. Hit Run next to the imported plugin:
image

## Links

***required*** - *Links to resources related to this Pull Request*

### CODE REVIEW

**Review by chrisadubois — APPROVED** [review]

File: packages/@momentum-design/figma-plugin-assets-export/src/plugin/index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+figma.showUI(__html__, { themeColors: true, height: 500, width: 500 });
+
+// Calls to "parent.postMessage" from within the HTML page will trigger this
+// callback. The callback will be passed the "pluginMessage" property of the
+// posted message.
+figma.ui.onmessage = (msg) => {
+  // used to distinguish between different post message types, sent from UI:
+  if (msg.type === 'export') {
+    // hard coded allowList, should be replaced by settings in UI:
+    const allowList = ['christoph', 'test'];
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/src/plugin/index.ts:14]: maybe change this? haha

File: packages/@momentum-design/figma-plugin-assets-export/src/plugin/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "compilerOptions": {
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/src/plugin/tsconfig.json:2]: document not formatted

File: packages/@momentum-design/figma-plugin-assets-export/manifest.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "Momentum Assets Export",
+  "id": "1180495970094217456",
+  "api": "1.0.0",
+  "main": "dist/index.js",
+  "editorType": [
+    "figma"
+  ],
+  "ui": "dist/index.html"
+}
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/manifest.json:10]: no new line

File: packages/@momentum-design/figma-plugin-assets-export/src/plugin/models/page.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable no-undef */
+class Page {
+    data: PageNode;
+
+    constructor(data: PageNode) {
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/src/plugin/models/page.ts:6]: not formatted

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/App.css [hunk 5]
```diff
@@ -0,0 +1,43 @@
+body {
+    font-family: Inter, Helvetica Neue, Helvetica, Arial, sans-serif;
+    color: var(--figma-color-text);
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/App.css:3]: not formatted

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/App.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    <div className="wrapper">
+      <Header/>
+      <div className="content">
+        <div className="form">
+          <FormRow>
+            <p className="assets-type-header">Assets type</p>
+          </FormRow>
+          <FormRow>
+            <Select name="assets-type" setSelectValue={setSelectedAssetType}>
+              <Option value="" disabled selected>Select assets type</Option>
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/App.tsx:25]: use placeholder on Select instead

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/components/Button/Button.css [hunk 7]
```diff
@@ -0,0 +1,22 @@
+button {
+    appearance: none;
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/components/Button/Button.css:2]: not formatted

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/components/Footer/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import './Footer.css';
+
+function Footer() {
+  return (
+    <div className="footer">
+      <p className="copyright">Made with ❤️ by <b>Momentum-Engineering</b></p>
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/components/Footer/Footer.tsx:8]: ❤️

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "target": "ESNext",
+        "useDefineForClassFields": true,
+        "lib": [
```
Comment (torgeadelin) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/tsconfig.json:5]: should probably be extended on the tsconfig of the monorepo

**Review by torgeadelin — APPROVED** [review]
Very small changes, approving since they are non blocking, would be nice to get them fixed tho. Loads of formatting issues, we really need to fix that eslint config!

**Pull request merged**